Sources/PackageRegistry/SignatureValidation.swift

```diff
@@ -201,6 +204,88 @@ struct SignatureValidation {
         }
     }
 
+    private func validateSourceArchiveSignature(
```

Comment:
This is the same as the original `validateSignature`.

Reply:
What's the reason for this function itself not being `async`? Only trying to understand whether we're trying to avoid `async` functions completely for now or is there some boundary along within which we should keep functions callback-based?